### PR TITLE
Feature/charging station sequences

### PR DESCRIPTION
**Pull request by thanaParis**
Moving from randomly generate requestIds to using a sequence table
Requests touched are updated to match latest format for custom actions.
requires changes in https://github.com/citrineos/citrineos-core/pull/283

image
image
image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -151,6 +155,11 @@ const hasuraOptions: HasuraDataProviderOptions = {
   },
 };
 
+const hasuraDataProvider = dataProvider(client, hasuraOptions)
+hasuraDataProvider.getApiUrl = () => {
```
Comment (elliot-sabitov) [inline src/App.tsx:159]: why is this needed? 
```
hasuraDataProvider.getApiUrl = () => {
  return API_URL;
}
```

File: src/message/get-base-report/index.tsx [hunk 2]
```diff
@@ -1,137 +1,132 @@
...
-    allValues: GetBaseReportRequest,
+
+  const getBaseReport = async (
+    request: GetBaseReportRequest,
   ) => {
-    setParentRecord(allValues);
+    try {
+      setLoading(true);
+      const client = new BaseRestClient();
+      const response = await client.post(
```
Comment (elliot-sabitov) [inline src/message/get-base-report/index.tsx:79]: if possible, can we please simplify this a bit by reusing this helper: 
```
await triggerMessageAndHandleResponse(
      `/configuration/triggerMessage?identifier=${stationId}&tenantId=1`,
      MessageConfirmation,
      data,
      (response: MessageConfirmation) => response && response.success,
    );
    ```

File: src/message/get-log/index.tsx [hunk 3]
```diff
@@ -1,154 +1,128 @@
...
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const [valid, setValid] = useState<boolean>(false);
+  
+  const apiUrl = useApiUrl();
+  const {
+    data: requestIdResponse,
+    isLoading: isLoadingRequestId,
+    // isError: isErrorLoadingRequestId,
+  } = useCustom<any>({
```
Comment (elliot-sabitov) [inline src/message/get-log/index.tsx:33]: please note, in this PR https://github.com/citrineos/citrineos-operator-ui/pull/25 we created the `useGqlCustom` helper method which will should more or less simplify this call to just be:
```
const {
    data: requestIdResponse,
    isLoading: isLoadingRequestId,
    // isError: isErrorLoadingRequestId,
  } = useCustom<any>({
      gqlQuery: CHARGING_STATION_SEQUENCES_GET_QUERY,
      variables: {
        stationId: station.id,
        type: 'getLog',
      },
  }
```

**Review by elliot-sabitov — APPROVED** [review]

**Comment by elliot-sabitov** [comment]
Please note: Build is failing because currently using CORE v1.4.3 and need changes for ChargingStationSequences to be merged into CORE rc-1.5.0 for it to be available. We may want to consider adjusting the build job to clone core locally to ensure that latest rc branch is used to ensure that it is able to build properly

File: src/App.tsx (hunk 1 above)
Comment (thanaParis) [reply src/App.tsx:159]: otherwise the method is not implemented.
